test/controllers/ConfirmRemoveWebsiteControllerSpec.scala

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import assets.BaseTestConstants._
+import models.User
+import models.customerInformation.UpdatePPOBSuccess
+import models.errors.ErrorModel
+import org.jsoup.Jsoup
+import play.api.http.Status
+import play.api.http.Status.{CONFLICT, INTERNAL_SERVER_ERROR}
+import play.api.mvc.AnyContent
+import play.api.test.Helpers._
+import views.html.ConfirmRemoveWebsiteView
+
+import scala.concurrent.Future
+
+class ConfirmRemoveWebsiteControllerSpec extends ControllerBaseSpec  {
+
+  object TestConfirmRemoveWebsiteController extends ConfirmRemoveWebsiteController(
+    mockAuthPredicateComponents,
+    mockInflightPPOBPredicate,
+    mcc,
+    mockErrorHandler,
+    mockVatSubscriptionService,
+    injector.instanceOf[ConfirmRemoveWebsiteView],
+    mockConfig
+  )
+
+  "Calling the extractWebsite function in ConfirmWebsiteController" when {
+
+    "there is an authenticated request from a user with an website address in session" should {
+
+      "result in an website address being retrieved if there is an website" in {
+        mockIndividualAuthorised()
+        val user = User[AnyContent](vrn, active = true, None)(requestWithWebsite)
+
+        TestConfirmRemoveWebsiteController.extractSessionWebsiteAddress(user) shouldBe Some(testWebsite)
+      }
+    }
+  }
+
+  "Calling the show action in ConfirmWebsiteController" when {
+
+    "there is an website address in session" should {
+
+      "show the Confirm website page" in {
+        mockIndividualAuthorised()
+        val result = TestConfirmRemoveWebsiteController.show(requestWithWebsite)
+
+        status(result) shouldBe Status.OK
+      }
+    }
+
+    "there isn't an website address in session" should {
+
+      "take the user to enter a new website address" in {
+        mockIndividualAuthorised()
+        val result = TestConfirmRemoveWebsiteController.show(request)
+
+        status(result) shouldBe Status.SEE_OTHER
+        redirectLocation(result) shouldBe Some("/vat-through-software/account/correspondence/new-website-address")
```

Comment:
Could you change this to be the reverse controller route instead of a hardcoded String as it's a bit more 'safe' and demonstrates the relationship between controllers better